edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java

```diff
@@ -3185,7 +3185,10 @@ private Double calculateCategoryScore(String studentUuid, Long categoryId, List<
 			Assignment assignment = gradeRecord.getAssignment();
 						
 			// remove if not for this category (rule 1)
-			if(assignment.getCategory() != null && categoryId.longValue() != assignment.getCategory().getId().longValue()){
+			if(assignment.getCategory() == null){
+				return true;
+			}
+			if(categoryId.longValue() != assignment.getCategory().getId().longValue()){
```

Comment:
longValue() returns a long. Both categoryID and assignemnt.getCategory are not null, so we then have a long != long comparison. So I believe that they would be equality correct, that is this statement and `categoryId.equals(assignment.getCategory().getId())` should be equivalent.